main/src/ca/uwaterloo/flix/language/ast/Kind.scala

```diff
@@ -39,6 +39,18 @@ sealed trait Kind {
     */
   override def toString: String = Kind.ShowInstance.show(this)
 
+  /**
+    * Returns true if `left` is a subkind of `this`.
+    * Right-associative.
+    */
+  def <::(left: Kind): Boolean = (left, this) match {
+    case (Kind.Record, Kind.Star) => true
+    case (Kind.Schema, Kind.Star) => true
+    case (Kind.Arrow(k11, k12), Kind.Arrow(k21, k22)) => (k11 <:: k21) && (k12 <:: k22)
+    case _ if (left == this) => true
```

Comment:
This case makes me uneasy. Let us just add (Star, Star), (Record, Record), and so forth.